api/api_public_transaction_pool.go

```diff
@@ -149,6 +150,48 @@ func (s *PublicTransactionPoolAPI) GetTransactionByHash(ctx context.Context, has
 	return nil
 }
 
+// GetDecodedAnchoringTransactionByHash returns the decoded anchoring data of anchoring transaction for the given hash
+func (s *PublicTransactionPoolAPI) GetDecodedAnchoringTransactionByHash(ctx context.Context, hash common.Hash) (map[string]interface{}, error) {
+	var tx *types.Transaction
+
+	// Try to return an already finalized transaction
+	if tx, _, _, _ = s.b.ChainDB().ReadTxAndLookupInfo(hash); tx != nil {
+		goto decode
+	}
+
+	// No finalized transaction, try to retrieve it from the pool
```

Comment:
Is it really necessary? I think that checking chainDB is enough.

Reply:
I just copied this code from `GetTransactionByHash` at line:154. :)
I think this API is similar with `GetTransactionByHash` API. So I think it's ok. What do you think?